shell/platform/windows/win32_window.cc

```diff
@@ -109,10 +110,70 @@ void Win32Window::TrackMouseLeaveEvent(HWND hwnd) {
   }
 }
 
+void Win32Window::OnImeSetContext(UINT const message,
+                                  WPARAM const wparam,
+                                  LPARAM const lparam) {
+  if (wparam != 0) {
+    text_input_manager_.CreateImeWindow(window_handle_);
+  }
+}
+
+void Win32Window::OnImeStartComposition(UINT const message,
+                                        WPARAM const wparam,
+                                        LPARAM const lparam) {
+  text_input_manager_.CreateImeWindow(window_handle_);
+  OnComposeBegin();
+}
+
+void Win32Window::OnImeComposition(UINT const message,
+                                   WPARAM const wparam,
+                                   LPARAM const lparam) {
+  // Update the IME window position.
+  text_input_manager_.UpdateImeWindow(window_handle_);
+
+  if (lparam & GCS_COMPSTR) {
+    // Read the in-progress composing string.
+    long pos = text_input_manager_.GetComposingCursorPosition(window_handle_);
+    std::u16string text{};
+    if (text_input_manager_.GetComposingString(window_handle_, &text)) {
+      OnComposeChange(text, pos);
+    }
+  } else if (lparam & GCS_RESULTSTR) {
+    // Read the committed composing string.
+    long pos = text_input_manager_.GetComposingCursorPosition(window_handle_);
+    std::u16string text{};
+    if (text_input_manager_.GetResultString(window_handle_, &text)) {
+      OnComposeChange(text, pos);
+    }
+    // Next, try reading the composing string. Some Japanese IMEs send a message
+    // containing both a GCS_RESULTSTR and a GCS_COMPSTR when one composition is
+    // committed and another immediately started.
+    if (text_input_manager_.GetComposingString(window_handle_, &text)) {
+      OnComposeChange(text, pos);
+    }
+  }
+}
+
+void Win32Window::OnImeEndComposition(UINT const message,
+                                      WPARAM const wparam,
+                                      LPARAM const lparam) {
+  text_input_manager_.DestroyImeWindow(window_handle_);
+  OnComposeEnd();
+}
+
+void Win32Window::OnImeRequest(UINT const message,
+                               WPARAM const wparam,
+                               LPARAM const lparam) {
+  // TODO(cbracken): Handle IMR_RECONVERTSTRING
+  // TODO(cbracken): Handle IMR_DOCUMENTFEED
+  // TODO(cbracken): Handle IMR_QUERYCHARPOSITION
```

Comment:
It would be good to file an issue to track this, capturing the details you mentioned in the PR description, and referring to it here.

Reply:
Done and referenced in the TODO. https://github.com/flutter/flutter/issues/74547